traitlets/traitlets.py

```diff
@@ -319,6 +320,23 @@ class that does the following:
        accept superclasses for :class:`This` values.
     """
 
+    def instance_init(self, obj):
```

Comment:
This is good, thanks. I know it's an instance attribute, but it's a nice convention to define instance attributes as class attributes, as a sort of minimal documentation of the interface.